uvicorn/workers.py

```diff
@@ -76,9 +77,22 @@ def init_signals(self) -> None:
         # Don't let SIGUSR1 disturb active requests by interrupting system calls
         signal.siginterrupt(signal.SIGUSR1, False)
 
+    def _install_sigquit_handler(self) -> None:
+        """Workaround to install a SIGQUIT handler on workers.
+        - https://github.com/encode/uvicorn/issues/1116
+        - https://github.com/benoitc/gunicorn/issues/2604
+        """
+        if threading.current_thread() is not threading.main_thread():
+            # Signals can only be listened to from the main thread.
+            return
```

Comment:
This shouldn't be needed. It's on the `Server` because `uvicorn` can be started in a thread, but the `UvicornWorker` cannot.